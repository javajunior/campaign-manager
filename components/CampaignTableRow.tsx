import React from 'react'
import numbro from 'numbro'
import { Table } from 'semantic-ui-react'

// Single row for the campaign table
const CampaignTableRow = ({ element, isActive }) => (
  <Table.Row>
    <Table.Cell>{element.id}</Table.Cell>
    <Table.Cell>{element.name}</Table.Cell>
    <Table.Cell>{element.startDate} </Table.Cell>
    <Table.Cell>{element.endDate}</Table.Cell>
    <Table.Cell negative={!isActive} positive={isActive}>
      {!isActive ? 'Ina' : 'A'}ctive
    </Table.Cell>
    <Table.Cell>
      {/* Use the numbro package to format the budget numbers */}
      {numbro(element.Budget).format({
        average: true,
        mantissa: 2,
      })}
    </Table.Cell>
  </Table.Row>
)

export default CampaignTableRow
