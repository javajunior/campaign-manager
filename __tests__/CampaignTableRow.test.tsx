import * as React from 'react'
import CampaignTableRow from '../components/CampaignTableRow'
import renderer from 'react-test-renderer'

it('renders the "CampaignTableRow" component correctly', () => {
  const tree = renderer
    .create(
      <CampaignTableRow
        element={{
          id: 1,
          name: 'Divavu',
          startDate: '9/19/2017',
          endDate: '3/9/2018',
          Budget: 88377,
        }}
        isActive={true}
      />,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
