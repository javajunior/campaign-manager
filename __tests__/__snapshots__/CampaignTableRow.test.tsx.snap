// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the "CampaignTableRow" component correctly 1`] = `
<tr
  className=""
>
  <td
    className=""
  >
    1
  </td>
  <td
    className=""
  >
    Divavu
  </td>
  <td
    className=""
  >
    9/19/2017
     
  </td>
  <td
    className=""
  >
    3/9/2018
  </td>
  <td
    className="positive"
  >
    A
    ctive
  </td>
  <td
    className=""
  >
    88.38k
  </td>
</tr>
`;
